libs/deposits/src/lib/use-allowance.ts

```diff
@@ -0,0 +1,28 @@
+import type { ERC20Token } from '@vegaprotocol/smart-contracts-sdk';
+import { useWeb3React } from '@web3-react/core';
+import type BigNumber from 'bignumber.js';
+import { useEffect, useState } from 'react';
+
+export const useAllowance = (
+  contract: ERC20Token | null,
+  bridgeAddress: string
+) => {
+  const { account } = useWeb3React();
+  const [allowance, setAllowance] = useState<BigNumber | null>(null);
+
+  useEffect(() => {
+    const run = async () => {
+      if (!contract || !account) {
+        return;
+      }
+
+      const res = await contract.allowance(account, bridgeAddress);
+
+      setAllowance(res);
+    };
+
+    run();
+  }, [contract, account, bridgeAddress]);
+
```

Comment:
chore (non-blocking): Will cause unmounted errors in race condition cases (same for all of below) should we have a `use-async` primitive or similar to prevent unmounted errors (see use fetch hook in block explorer for example. Also use fetch should move into react helpers)

Reply:
I've added a use-ethereum-read-contract hook, which looks very similar to use-fetch except it takes a function as an argument rather than a url and of course doesn't do any response handling.